src/checkers/inference/model/ConstraintManager.java

```diff
@@ -32,13 +33,13 @@
 
     private SourceChecker checker;
 
-    private QualifierHierarchy qualHierarchy;
+    private QualifierHierarchy realQualHierarchy;
 
     private VisitorState visitorState;
 
-    public void init(InferenceAnnotatedTypeFactory inferenceTypeFactory) {
+    public void init(InferenceAnnotatedTypeFactory inferenceTypeFactory, AnnotatedTypeFactory realTypeFactory) {
```

Comment:
That's a good suggestion and would allow us to access the real ATF in other locations.
The other changes below look good - one must use the real ATF when comparing constants.